src/elm/Feature.elm

```diff
@@ -0,0 +1,29 @@
+module Feature exposing (Feature(..), FeatureFlags(..), add, has, remove)
+
+
+type FeatureFlags
+    = FeatureFlags (List Feature)
+
+
+type Feature
+    = Shop
+    | Actions
```

Comment:
This isn't expressed in the type system, but the `add` function checks if the Feature being added is already in the list before doing so. I originally had a record with booleans like you said, but I personally think it's clearer to have a single type for features, even if both approaches are functionally the same.

Reply:
+1 to @amiskov point. Since you are working with a list, you may get more elements than required. If you want to keep a list its best to create a custom list, restricted to a given length and a uniqueness to its type.

Lists are not very good in this case because in order to know if a community has a feature, you would have to transverse the list.

It would be ok to have a a `Set` (https://package.elm-lang.org/packages/elm/core/latest/Set). with tuples into it. 

I again, too much of a stretch just to have a list.

And just for curiosity, are you trying to use lists to have some kind of dynamic building of the interface as the API add new fields?

Reply:
because if you are trying to achieve HATEOAS, which is of course doable, we should change our whole app strategy (maybe build a new one), applying HATEOAS concepts here will only make it difficult

Reply:
Changed on the latest commit :)